### PR TITLE
[CELEBORN-787] Add chunk related UTs for FileWriter

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
Add chunk related UTs for FileWriter.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Passes GA.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @RexXiong @pan3793 @AngersZhuuuu @cfmcgrady

**Review by pan3793 — APPROVED** [review]

**Review by AngersZhuuuu — APPROVED** [review]

**Review by RexXiong — APPROVED** [review]
+1, Thanks!

**Review by cfmcgrady — APPROVED** [review]
LGTM. but it could not cover https://github.com/apache/incubator-celeborn/pull/1702 changes?

**Comment by waitinfuture** [comment]
> 

Good point! Also @AngersZhuuuu reminded me about that. I added more cases, and without https://github.com/apache/incubator-celeborn/pull/1702 the UT will fail:
image

File: worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/FileWriterSuiteJ.java [hunk 1]
```diff
@@ -679,7 +679,7 @@ public void testChunkSize() throws IOException {
     assertEquals(fileInfo.getLastChunkOffset(), 4096);
     assertEquals(fileInfo.getChunkOffsets().get(3) - fileInfo.getChunkOffsets().get(2), 2048);
 
-    // case 7.2: write 4097B with 3 chunks with trim
+    // case 7.2: write 4097B with 3 chunks with trim; Without [CELEBORN-626][FOLLOWUP] this case will fail
```
Comment (pan3793) [inline worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/FileWriterSuiteJ.java:682]:
```suggestion
    // case 7.2: write 4097B with 3 chunks with trim; without PR #1702 this case will fail
```

**Pull request closed without merging**